environs/bootstrap/bootstrap.go

```diff
@@ -362,7 +362,10 @@ func bootstrapIAAS(
 			args.BootstrapBase = base
 			logger.Debugf("auto-selecting bootstrap series %q", args.BootstrapBase.String())
 		}
-		if args.BootstrapConstraints.Arch == nil && args.ModelConstraints.Arch == nil && detectedHW.Arch != nil {
+		if args.BootstrapConstraints.Arch == nil &&
+			args.ModelConstraints.Arch == nil &&
+			detectedHW != nil &&
+			detectedHW.Arch != nil {
 			arch := *detectedHW.Arch
 			args.BootstrapConstraints.Arch = &arch
```

Comment:
We could set the ModelConstraints.Arch (if not set). The hw detector could return an extra bool indicating whether the detected hw should be bootstrap only. The manual provider would be true.

Reply:
I ended up adding a new method on the detector, because I think it provides a better separation of concerns. Is this OK with you?